server/src/main/java/org/elasticsearch/index/seqno/ReplicationTracker.java

```diff
@@ -292,7 +291,7 @@ public synchronized RetentionLease renewRetentionLease(final String id, final lo
      * @param id       the identifier of the retention lease
      * @param listener the callback when the retention lease is successfully removed and synced to replicas
      */
-    public void removeRetentionLease(final String id, final ActionListener<ReplicationResponse> listener) {
+    public void removeRetentionLease(final String id, final ActionListener<Void> listener) {
```

Comment:
I know we are not really using the response but is there any reason for this change? Is it okay if we make both sync and backgroundSync return ReplicationResponse instead of Void?

Reply:
Primarily because I think `RetentionLeaseSyncer.EMPTY` should notify the listener (the contract for a method that accepts an `ActionListener` is that it is eventually notified, which today we do not satisfy) and yet that implementation does not have a `ReplicationResponse` with which to notify it. I could make one up, but it felt wrong.

Reply:
I think we can move `RetentionLeaseSyncer.EMPTY` to tests and make it notify listeners with an empty ReplicationResponse?

Reply:
Ok, I reverted back to using a made-up `ReplicationResponse` in 50acab784f4f9c7bb7090ba74676b51e4da6e1a5; I think I'll leave `RetentionLeaseSyncer.EMPTY` where it is.